core/src/main/java/org/infinispan/persistence/manager/PersistenceManager.java

```diff
@@ -51,6 +51,25 @@
     */
    CompletionStage<Void> disableStore(String storeType);
 
+   /**
+    * Adds a new store to the cache.
+    *
+    * @param storeConfiguration the configuration for the store
+    * @throws org.infinispan.commons.CacheException if the cache is was not empty
+    */
+   CompletionStage<Void> addStore(StoreConfiguration storeConfiguration);
+
+   /**
+    * Add a {@link StoreChangeListener} to be notified when a store is added or removed dynamically.
+    */
+   void addStoreListener(StoreChangeListener listener);
+
+   void removeStoreListener(StoreChangeListener listener);
+
+   interface StoreChangeListener {
+      void storeChanged(PersistenceStatus persistenceStatus);
```

Comment:
It kinda needs additional synchronization depending on the implementation, it can be as simple as using volatiles, or extra locks to guarantee atomicity with multiple operations.